lib/teleterm/apiserver/handler/handler_auth.go

```diff
@@ -83,6 +88,11 @@ func (s *Handler) LoginPasswordless(stream api.TerminalService_LoginPasswordless
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	if !cluster.URI.IsRoot() {
+		return trace.BadParameter("cluster URI must be a root URI")
```

Comment:
That's fine for now, but in a completely new code it'd have been better to convert the string into `uri.ResourceURI`, then check if it points to a root cluster and only then call `s.DaemonService.ResolveClusterURI`.